File: src/types/analytics.ts
import { z } from 'zod'

export const monthlySpendSchema = z.object({
  month: z.string(),
  total: z.number(),
  count: z.number(),
})

export const categorySpendSchema = z.object({
  category_id: z.string().nullable(),
  category_name: z.string(),
  total: z.number(),
  percentage: z.number(),
})

export const yearSummarySchema = z.object({
  year: z.number(),
  total_spent: z.number(),
  avg_monthly: z.number(),
  highest_month: z.string(),
  highest_amount: z.number(),
  category_breakdown: z.array(categorySpendSchema),
})

export const spendingVelocitySchema = z.object({
  current_month: z.number(),
  previous_month: z.number(),
  change_percent: z.number(),
  projected_annual: z.number(),
})

export type MonthlySpend = z.infer<typeof monthlySpendSchema>
export type CategorySpend = z.infer<typeof categorySpendSchema>
export type YearSummary = z.infer<typeof yearSummarySchema>
export type SpendingVelocity = z.infer<typeof spendingVelocitySchema>

File: crates/subby-core/src/models/analytics.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct MonthlySpend {
    pub month: String,
    pub total: f64,
    pub count: i32,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategorySpend {
    pub category_id: Option<String>,
    pub category_name: String,
    pub total: f64,
    pub percentage: f64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct YearSummary {
    pub year: i32,
    pub total_spent: f64,
    pub avg_monthly: f64,
    pub highest_month: String,
    pub highest_amount: f64,
    pub category_breakdown: Vec<CategorySpend>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SpendingVelocity {
    pub current_month: f64,
    pub previous_month: f64,
    pub change_percent: f64,
    pub projected_annual: f64,
}
